modules/k8s/pod.go

```diff
@@ -142,7 +142,7 @@ func IsPodAvailable(pod *corev1.Pod) bool {
 		isContainerStarted := containerStatus.Started
 		isContainerReady := containerStatus.Ready
 
-		if !isContainerReady && isContainerStarted != nil && !*isContainerStarted {
+		if !isContainerReady || isContainerStarted == nil || !*isContainerStarted {
```

Comment:
NIT: `*isContainerStarted == false` is a tiny bit easier to read than `!*isContainerStarted`.